src/Raven.Server/Documents/Sharding/Documents/ShardedSubscriptionContext.cs

```diff
@@ -0,0 +1,201 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Raven.Client.Documents.Commands;
+using Raven.Client.Documents.Subscriptions;
+using Raven.Client.Exceptions.Database;
+using Raven.Client.Exceptions.Documents.Subscriptions;
+using Raven.Client.Json.Serialization;
+using Raven.Server.Documents.Subscriptions;
+using Raven.Server.Rachis;
+using Raven.Server.ServerWide;
+using Raven.Server.ServerWide.Context;
+using Sparrow.Json;
+using Sparrow.Logging;
+
+namespace Raven.Server.Documents.Sharding.Documents
+{
+    public class ShardedSubscriptionContext : SubscriptionStorageBase, IDisposable /*, ILowMemoryHandler*/
```

Comment:
Why does the context inherit from storage?

Reply:
this is sharded subscription storage, it inherit from `SubscriptionStorageBase` because I use most of its methods for sharded subscription as well. its named context just for easier disguise between the sharded and not shared one.